### PR TITLE
Fix reference issue where paramname is referred instead of basename

### PR DESCRIPTION
**Pull request by AsabuHere**
# Closes #
[DII-782](https://issues.corp.twilio.com/browse/DII-782) Broken CreateParticipantOptions parameters in v6 of twilio-csharp library
A short description of what this PR does.
In previous versions of the C# library the MessagingBindingAddress and MessagingBindingProxyAddress properties for the CreateParticipantOptions class would translated into MessagingBinding.Address and MessagingBinding.ProxyAddress parameters before submitting to the REST API.
However, in v6 it is now MessagingBindingAddress and MessagingBindingProxyAddress which are ignore by the REST API.

[Code v5](https://github.com/twilio/twilio-csharp/blob/5.81.2/src/Twilio/Rest/Conversations/V1/Conversation/ParticipantOptions.cs#L80-L88) vs [Code v6](https://github.com/twilio/twilio-csharp/blob/main/src/Twilio/Rest/Conversations/V1/Conversation/ParticipantOptions.cs#L79-L86).

It should be MessagingBinding.Address and MessagingBinding.ProxyAddress or it won't work. I'm not sure if there are other APIs that have similar dot-notations that are affected in v6.
### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [X] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

**Review by childish-sambino — COMMENTED** [review]
Change looks fine, but need to generate the corresponding PR for twilio-csharp and link it and the GitHub issue here.

File: src/main/resources/twilio-csharp/options/GetParams.mustache [hunk 1]
```diff
@@ -15,7 +15,7 @@
             {{^isMap}}
             if ({{paramName}} != null)
             {
-                p.Add(new KeyValuePair<string, string>("{{paramName}}", {{{vendorExtensions.x-param-to-string}}}));
+                p.Add(new KeyValuePair<string, string>("{{{baseName}}}", {{{vendorExtensions.x-param-to-string}}}));
```
Comment (sbansla) [inline src/main/resources/twilio-csharp/options/GetParams.mustache:18]: This change needs to be done in case of map and array as well.

**Review by sbansla — CHANGES_REQUESTED** [review]

**Comment by AsabuHere** [comment]
> Change looks fine, but need to generate the corresponding PR for twilio-csharp and link it and the GitHub issue here.

https://github.com/twilio/twilio-csharp/pull/657

**Review by sbansla — APPROVED** [review]
Changes looks fine.

**Pull request merged**